Mathlib/Topology/MetricSpace/PiNat.lean

```diff
@@ -48,6 +48,29 @@ in general), and `ι` is countable.
   the uniformity is definitionally the product uniformity. Not registered as an instance.
 -/
 
+namespace PseudoMetricSpace
+variable {X : Type*}
+
+/-- Build a new pseudometric space from an old one where the distance uniform structure is provably
+(but typically non-definitionally) equal to some given distance structure. -/
+-- See note [forgetful inheritance]
+-- See note [reducible non-instances]
+abbrev replaceDist (m : PseudoMetricSpace X) (d : X → X → ℝ) (hd : d = dist) :
+    PseudoMetricSpace X where
```

Comment:
1. We should have a version of this for `PseudoEMetricSpace` that replaces `edist`.
2. I think this `replaceDist` should actually replace both `dist` and `edist` simultaneously.
3. It might also make sense to have convenience versions for `MetricSpace` and `EMetricSpace` too.
4. Please add some `example`s below this which guarantee that the forgetfully inherited structures (bornology and uniformity) are unchanged from the original at `by with_reducible` transparency. Even if it already works in this PR (which I think it should), the purpose is to make sure it stays that way.

Reply:
But it is also trivial AND more modular to replace dist and edist by composing two functions. If anything, the function you're suggesting should be called `replaceDistEDist` if it existed, and I claim it should be an abbrev for `replaceDist` composed with `replaceEDist`